File: client/src/models/Restaurant.js
import { APIItem } from "./APIItem.js"
import { Profile } from "./Profile.js"

export class Restaurant extends APIItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.imgUrl = data.imgUrl
    this.description = data.description
    this.visits = data.visits
    this.isShutdown = data.isShutdown
    this.creatorId = data.creatorId
    this.owner = new Profile(data.owner)
    this.reportCount = data.reportCount
  }
}

File: server/Models/Restaurant.cs
using System.ComponentModel.DataAnnotations;

namespace help_reviews.Models;

public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  [Url, MaxLength(3000)] public string ImgUrl { get; set; }
  [MaxLength(2000)] public string Description { get; set; }
  public int Visits { get; set; }
  public bool? IsShutdown { get; set; } //nullable
  public string CreatorId { get; set; }
  public Profile Owner { get; set; }
  public int ReportCount { get; set; }
}